rts/System/Sync/DumpState.cpp

```diff
@@ -109,6 +114,77 @@ namespace {
 			p[8], p[9], p[10], p[11], p[12], p[13], p[14], p[15]
 		);
 	}
+
+	template<typename T>
+	concept StringLike = std::convertible_to<std::remove_cvref_t<T>, std::string_view>;
+
+	template<typename T>
+	concept SequenceContainer = std::ranges::range<T>
+		&& !StringLike<T>
+		&& !requires { typename T::key_type; };
+
+	template<typename T>
+	concept SetLikeContainer = requires { typename T::key_type; }
+	&& !requires { typename T::mapped_type; };
+
+	template<typename T>
+	concept MapLikeContainer = requires {
+		typename T::key_type;
+		typename T::mapped_type;
+	};
+
+	template<typename T>
+	concept Numeric = std::integral<T> || std::floating_point<T>;
+
+	//template<typename T>
+	//inline uint32_t CheckSum(T&& r, uint32_t cs = 0u)
+	//{
+	//	static_assert(false, "Not Implemented");
```

Comment:
uncomment or remove? (or maybe it serves as a reminder but would become used due to SFINAE if uncommented? in that case sounds ok to keep as-is)